src/common/com/intellij/plugins/haxe/ide/refactoring/introduce/HaxeIntroduceHandler.java

```diff
@@ -148,14 +148,24 @@ protected void performAction(HaxeIntroduceOperation operation) {
         element2 = file.findElementAt(document.getLineEndOffset(lineNumber) - 1);
       }
     }
+    if (element1 instanceof PsiWhiteSpace) {
+      int startOffset = element1.getTextRange().getEndOffset();
+      element1 = file.findElementAt(startOffset);
+    }
+    if (element2 instanceof PsiWhiteSpace) {
+      int endOffset = element2.getTextRange().getStartOffset();
+      element2 = file.findElementAt(endOffset - 1);
+    }
+
+
     final Project project = operation.getProject();
     if (element1 == null || element2 == null) {
       showCannotPerformError(project, editor);
       return;
     }
 
     element1 = HaxeRefactoringUtil.getSelectedExpression(project, file, element1, element2);
-    if (element1 == null) {
+    if (element1 == null || shouldExclude(element1)) {
```

Comment:
Could it be isValidForExtraction?
`if (element == null || !isValidForExtraction(element1))`

Reply:
Sure.  Done.